docs/react-v9/contributing/rfcs/react-components/convergence/z-index-handling.md

```diff
@@ -0,0 +1,319 @@
+  floating: 4000, // floating elements - dropdowns, teaching
+  priority: 5000, // priority elements - tooltips
+  debug: 6000, // debug - error overlays, debug messages
+};
+```
+
+```ts
+// partner side
+
+import { zIndexes } from '@fluentui/theme';
+
+const styles = {
+  root: {
+    zIndex: zIndexes.overlay,
+  },
+};
+```
+
+<!-- FIXME: Update section after alignment with designers -->
+
+_NOTE: The names above are not final and only serve as an example. As of the moment this RFC was introduced, there are ongoing discussions with the Design team to define all the layers. This RFC will be updated before the final approval to include the definitive names._
+
+##### Pros
+
+- 👍 Clear separation by group of UI elements
+- 👍 Easy to override, extend and update
+
+##### Cons
+
+- 👎 Naming convention can be hard
+- 👎 The values would be fixed globally and not possible for partners to override
+
+#### Option B: Update tokens to include a map from layer to z-index
+
+Expose the z-index values as tokens. This would map the layers into token values that can be used anywhere in styles.
+
+The layers can be defined as follows:
+
+```ts
+// packages/tokens/src/global/zIndexes.ts
+
+import { ZIndexLevels, ZIndexTokens } from '../types';
+
+// According to Fluent V2 guidelines
+const levels: ZIndexLevels = [
+  0, // Elevation 0
+  1, // Elevation 2
+  1000, // Elevation 4
+  2000, // Elevation 8
+  3000, // Elevation 16
+  4000, // Elevation 28
+  5000, // Elevation 64
+  6000, // High priority elements
+];
+
+/**
+ * Global z-index values for elements
+ */
+export const zIndexes: ZIndexTokens = {
+  zIndexBackground: levels[0], // default
+  zIndexContent: levels[1], // content - header, footer, sidebar
+  zIndexOverlay: levels[2], // overlay elements - drawer, nav
+  zIndexPopup: levels[3], // popup layers - popups, modals, dialogs
+  zIndexMessages: levels[4], // communication elements - banners, messages, toasts, snackbar
+  zIndexFloating: levels[5], // floating elements - dropdowns, teaching
+  zIndexPriority: levels[6], // priority elements - tooltips
+  zIndexDebug: levels[7], // debug - error overlays, debug messages
+};
+```
+
+```ts
+// partner side
+
+import { tokens } from '@fluentui/theme';
+
+const styles = {
+  root: {
+    zIndex: tokens.zIndexOverlay,
+  },
+};
+```
+
+<!-- FIXME: Update section after alignment with designers -->
+
+_NOTE: The names above are not final and only serve as an example. As of the moment this RFC was introduced, there are ongoing discussions with the Design team to define all the layers. This RFC will be updated before the final approval to include the definitive names._
+
+##### Pros
+
+- 👍 Clear separation by group of UI elements
+- 👍 Easy to override, extend and update
+- 👍 Partners can override the tokens through themes
+
+##### Cons
+
+- 👎 Naming convention can be hard
+- 👎 Bloating the tokens with more variables
+- 👎 Not possible to validate for user errors. Partners can override the tokens freely.
```

Comment:
I am unsure if this is an issue as it's the same for existing tokens, too

Reply:
This is an issue that happens for existing tokens, but since I wanted to emphasize that, for the z-index feature, this would also be the case.